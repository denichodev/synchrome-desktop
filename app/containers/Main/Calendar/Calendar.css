.calendarContainer {
  overflow: hidden;
}

.pullheight {
  height: 100%;
}

.fixBotMargin {
  margin-bottom: 24px;  
}
