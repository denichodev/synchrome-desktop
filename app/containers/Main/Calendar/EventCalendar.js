import React, { Component } from 'react';
import $ from 'jquery';
import 'moment';
import 'fullcalendar/dist/fullcalendar';
import styles from './EventCalendar.css';

class EventCalendar extends Component {

  static defaultProps = {
    defaultView: 'month',
    displayEventTime: 'false',
    selectable: false,
    handleSelection: (start, end) => {
      console.log(start, end);
    }
  }

  componentDidMount() {
    const { defaultView, height, header, events, displayEventTime, selectable, handleSelection } = this.props;

    $('#calendar').fullCalendar({
      defaultView,
      height,
      header,
      events,
      displayEventTime,
      selectable,
      select: handleSelection
    });
  }

  render() {
    return (
      <div id="calendar" className={styles.fullCalendar} />
    );
  }
}

export default EventCalendar;
