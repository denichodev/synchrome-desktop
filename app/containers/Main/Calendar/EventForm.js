import React, { Component } from 'react';
import { Field, reduxForm } from 'redux-form';

class EventForm extends Component {
  renderTextField = field => {
    // const { meta: { touched, error } } = field;
    // const className = `form-group ${touched && error ? 'has-danger' : ''}`;
    const { label, name, input, placeholder } = field;
    return (
      <div className="form-group">
        <label htmlFor={name}>
          {label}
        </label>
        <input
          type="text"
          className="form-control"
          placeholder={placeholder}
          {...input}
        />
      </div>
    );
  };

  renderForm = () => (
    <form>
      <Field
        label="Event Name"
        name="name"
        placeholder="Enter your event name here"
        component={this.renderTextField}
      />
      <Field
        label="Start date"
        name="start"
        placeholder="Enter start date"
        component={this.renderTextField}
      />
      <Field
        label="End date"
        name="end"
        placeholder="Enter end date"
        component={this.renderTextField}
      />
    </form>
  );

  render() {
    return (
      <div className="animated fadeIn">
        <div className="row">
          <div className="col-md-12 col-sm-12">
            <div className="card">
              <div className="card-header">Add New Event</div>
              <div className="card-block">
                {this.renderForm()}
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const formOptions = {
  form: 'EventsNewForm',
  destroyOnUnmount: false
};

export default reduxForm(formOptions)(EventForm);
