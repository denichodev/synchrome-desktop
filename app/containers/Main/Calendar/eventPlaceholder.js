const randomDay = () => {
  return Math.floor((Math.random() * 28) + 1).toString();
}

const randomMonth = () => {
  return Math.floor((Math.random() * 12) + 1).toString();
}

export const generateEvents = () => {
  let arr = [];
  for (let i = 0; i < 500; i++) {
    arr.push({
      title: `event ${i}`,
      start: `2017-${randomMonth()}-${randomDay()}`,
      color: 'red'
    });
  }

  arr.push({
    title: 'event setahun',
    start: '2017-01-01',
    end: '2017-12-31'
  });

  return arr;
};
