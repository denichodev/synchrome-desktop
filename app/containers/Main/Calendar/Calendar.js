// @flow
import React, { Component } from 'react';

import EventCalendar from './EventCalendar';

class Calendar extends Component {
  render() {
    return (
      <div className="animated fadeIn">
        <div className="col-md-12">
          <div className="card">
            <div className="card-block">
              <EventCalendar
                height={530}
              />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Calendar;
