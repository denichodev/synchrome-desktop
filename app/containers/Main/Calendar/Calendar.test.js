import React from 'react';
import configureStore from 'redux-mock-store';
import Calendar from './Calendar';

const mockStore = configureStore();

describe('Calendar', () => {
  it('should match its snapshot', () => {
    const component = shallow(
      <Calendar store={mockStore()} />
    );

    expect(component).toMatchSnapshot();
  });

  it('normalize the fetched calendar events', () => {
    const mockCalendar = {
      id: '1',
      name: 'Mock Calendar',
      events: [
        {
          id: 1,
          title: 'Hari Kemerdekaan',
          start: '2017-08-17',
          end: '2017-08-17',
          category: { color: '#bada55', textColor: '#111'}
        },
        {
          id: 2,
          title: 'Tahun Baru',
          start: '2017-01-01',
          end: '2017-01-01',
          category: { color: '#efefef', textColor: '#111'}
        }
      ]
    }

    const expectedEvents = [
      {
        id: 1,
        title: 'Hari Kemerdekaan',
        start: '2017-08-17',
        end: '2017-08-17',
        color: '#bada55',
        textColor: "111"
      },
      {
        id: 2,
        title: 'Tahun Baru',
        start: '2017-01-01',
        end: '2017-01-01',
        color: '#efefef',
        textColor: "111"
      }
    ];

    const wrapper = shallow(<Calendar store={mockStore()} />);
    const instance = wrapper.dive();
  });
});
