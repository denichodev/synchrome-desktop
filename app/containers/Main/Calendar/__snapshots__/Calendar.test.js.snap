// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar should match snapshot 1`] = `
<div
  className="animated fadein"
>
  <div
    className="col-md-12"
  >
    <div
      className="card"
    >
      <div
        className="card-block"
      >
        <EventCalendar
          defaultView="month"
          height={530}
        />
      </div>
    </div>
  </div>
</div>
`;
